src/main/scala/sigmastate/serialization/ValueSerializer.scala

```diff
@@ -17,45 +18,38 @@ trait ValueSerializer[V <: Value[SType]] extends SigmaSerializer[Value[SType], V
   }
 }
 
-object ValueSerializer extends SigmaSerializerCompanion[Value[SType]] {
-  type OpCode = Byte
-  type Tag = OpCode
-
-  val TaggedVariableCode = 1: Byte
-
-  val IntConstantCode = 11: Byte
-  val TrueCode = 12: Byte
-  val FalseCode = 13: Byte
-  val UnitConstantCode = 14: Byte
-  val ByteArrayConstantCode = 15: Byte
-
-  val LtCode = 21: Byte
-  val LeCode = 22: Byte
-  val GtCode = 23: Byte
-  val GeCode = 24: Byte
-  val EqCode = 25: Byte
-  val NeqCode = 26: Byte
+object ValueSerializer
+  extends SigmaSerializerCompanion[Value[SType]] {
 
-  val ConcreteCollectionCode = 35: Byte
-  val TupleCode = 36: Byte
-  val AndCode = 37: Byte
-  val OrCode = 38: Byte
+  type Tag = OpCode
 
   val table = Seq[ValueSerializer[_ <: Value[SType]]](
+
     RelationSerializer(GtCode, GT.apply, Seq(Constraints.onlyInt2)),
     RelationSerializer(GeCode, GE.apply, Seq(Constraints.onlyInt2)),
     RelationSerializer(LtCode, LT.apply, Seq(Constraints.onlyInt2)),
     RelationSerializer(LeCode, LE.apply, Seq(Constraints.onlyInt2)),
     RelationSerializer(EqCode, EQ.applyNonTyped, Seq(Constraints.sameType2)),
     RelationSerializer(NeqCode, NEQ.apply, Seq(Constraints.sameType2)),
+
+    TwoArgumentsSerializer(XorCode, Xor.apply),
+    TwoArgumentsSerializer(AppendBytesCode, AppendBytes.apply),
+    TwoConstrainedArgumentsSerializer(ExponentiateCode, Exponentiate.apply, Seq(Constraints.sameType2)),
```

Comment:
Why same-type constraint here? Exponentiate parameters are group element and big int

Reply:
I removed TwoArgumentsSerializer